tests/_helpers/mock.zsh

```diff
@@ -0,0 +1,29 @@
+mock() {
+    local target=$1
+    shift
+
+    echo 0 > ${target}_mock_times
+
+    $target() {
```

Comment:
Not sure if I want to do so because it could not be generally possible to determine which argument is a subcommand when the mock function receives arguments from the caller.

Reply:
Ah, All it have to do is use `command git reset` in `@teardown` .

Reply:
This kind of technique which ‘proxies’ some of the invocations to a mock object is often referred to as partial mocking that I personally find it very confusing over its behaviour because one can hardly find whether an invocation goes to the real implementation or is just mocked. The ideal is to choose from: A) mock everything, or B) use the real implementation everywhere. Note the following:
- A local directory can technically be a remote repository which could potentially eliminate the entire need for mock here
- Mocked functions are reverted to the original state after calling `unmock`. `git` will work as a command once being unmocked.